src/main/java/duke/Duke.java

```diff
@@ -29,33 +31,49 @@ public void start() {
         while (true) {
             String input = sc.nextLine();
             echo(dispatch(input));
-            if (input.equals("bye"))
+            if (input.equals("bye")) {
+                this.sc.close();
                 break;
+            }
         }
     }
 
     private String dispatch(String input) {
         switch (input) {
-        case "list":
-            StringBuilder sb = new StringBuilder();
-            int tasksLength = this.tasks.size();
-            for (int i = 0; i < tasksLength - 1; i++) {
-                sb.append(String.format("%d. %s\n", i + 1, this.tasks.get(i)));
-            }
-            sb.append(String.format("%d. %s", tasksLength, this.tasks.get(tasksLength - 1)));
-            return sb.toString();
-        case "bye":
-            return "Bye. Hope to see you again soon!";
-        default:
-            if (Pattern.matches("^done\\s+\\d$", input)) {
-                int doneInd = Integer.parseInt((input.split(" "))[1]) - 1;
-                Task currTask = this.tasks.get(doneInd);
-                currTask.setDone();
-                return String.format("Nice! I've marked this task as done:\n  %s", currTask);
-            } else {
-                this.tasks.add(new Task(input));
-                return String.format("added: %s", input);
-            }
+            case "list":
+                int tasksLength = this.tasks.size();
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < tasksLength - 1; i++) {
+                    sb.append(String.format("%d. %s\n", i + 1, this.tasks.get(i)));
+                }
+                sb.append(String.format("%d. %s", tasksLength, this.tasks.get(tasksLength - 1)));
+                return sb.toString();
+            case "bye":
+                return "Bye. Hope to see you again soon!";
+            default:
+                if (Pattern.matches("^done\\s+\\d$", input)) {
+                    int doneInd = Integer.parseInt((input.split(" "))[1]) - 1;
+                    if (doneInd >= this.tasks.size()) {
```

Comment:
It might be clearer if you list down all the cases clearly instead of putting together under "default".